src/test/java/org/opentripplanner/middleware/controllers/api/OtpUserControllerTest.java

```diff
@@ -60,69 +61,70 @@ public static void tearDown() {
+        HttpResponse<String> otpUserWithPhoneRequest = mockAuthenticatedRequest(
             String.format("api/secure/user/%s", otpUser.id),
             otpUser,
             HttpUtils.REQUEST_METHOD.GET
         );
-        assertEquals(HttpStatus.OK_200, otpUserWithPhoneResponse.statusCode());
+        assertEquals(HttpStatus.OK_200, otpUserWithPhoneRequest.statusCode());
 
-        OtpUser otpUserWithPhone = JsonUtils.getPOJOFromJSON(otpUserWithPhoneResponse.body(), OtpUser.class);
+        OtpUser otpUserWithPhone = JsonUtils.getPOJOFromJSON(otpUserWithPhoneRequest.body(), OtpUser.class);
         assertEquals(INITIAL_PHONE_NUMBER, otpUserWithPhone.phoneNumber);
-        assertEquals(PHONE_NUMBER_TO_VERIFY, otpUserWithPhone.pendingPhoneNumber);
-        assertEquals(PHONE_NUMBER_TO_VERIFY_FORMATTED, otpUserWithPhone.pendingPhoneNumberFormatted);
+        assertTrue(otpUserWithPhone.isPhoneNumberVerified);
+    }
+
+    private static Set<Map.Entry<String, Integer>> createBadPhoneNumbers() {
+        HashMap<String, Integer> cases = new HashMap<>();
+        cases.put("5555555", HttpStatus.BAD_REQUEST_400);
+        cases.put("+15555550001", HttpStatus.INTERNAL_SERVER_ERROR_500);
+
+        return cases.entrySet();
     }
 
     /**
-     * Check that a request with an malformed number or an international number
-     * results in a 400-bad request response.
-     * The home country is set by the COUNTRY_CODE config parameter.
+     * Tests that phone numbers meet the E.164 format (e.g. +1555555).
      */
     @ParameterizedTest
-    @MethodSource("createRejectedNumbers")
-    public void invalidOrForeignNumbersShouldProduceBadRequest(String number) {
-        assumeTrue(NotificationUtils.COUNTRY_CODE.equals("US"));
+    @MethodSource("createPhoneNumberTestCases")
+    public void isPhoneNumberValidE164(Map.Entry<String, Boolean> testCase) {
+        assumeTrue(testsShouldRun());
+        String number = testCase.getKey();
+        boolean isValid = testCase.getValue();
 
-        HttpResponse<String> response = mockAuthenticatedRequest(
-            String.format("api/secure/user/%s/verify_sms/%s",
-                otpUser.id,
-                number
-            ),
-            otpUser,
-            HttpUtils.REQUEST_METHOD.GET
-        );
-        assertEquals(HttpStatus.BAD_REQUEST_400, response.statusCode());
+        assertEquals(isValid, OtpUserController.isPhoneNumberValidE164(number));
     }
 
-    private static List<String> createRejectedNumbers() {
-        return List.of(
-            // "Famous" old parisian print shop number (https://fr.wikipedia.org/wiki/Jean_Mineur)
-            "+33142250001",
+    private static Set<Map.Entry<String, Boolean>> createPhoneNumberTestCases() {
+        HashMap<String, Boolean> cases = new HashMap<>();
+        cases.put("+15555550123", true);
+        cases.put("+1 5555550123", false); // no spaces allowed.
+        cases.put("(555) 555,0123", false);
+        cases.put("555555", false);
```

Comment:
Same comment about Arguments.

Reply:
Good find, it's a cleaner syntax. Updated in https://github.com/ibi-group/otp-middleware/pull/83/commits/8ec525aabdaac519033d9918bca89ad483ecaa7f.